Python/ast.c

```diff
@@ -1576,10 +1576,11 @@ ast_for_decorators(struct compiling *c, const node *n)
 }
 
 static stmt_ty
-ast_for_funcdef_impl(struct compiling *c, const node *n,
+ast_for_funcdef_impl(struct compiling *c, const node *n0,
                      asdl_seq *decorator_seq, int is_async)
 {
     /* funcdef: 'def' NAME parameters ['->' test] ':' suite */
+    const node * const n = is_async == 0 ? n0 : CHILD(n0, 1);
```

Comment:
`is_async == 0` should be in parenthesis.

Alternatively, do you just want to make the `is_async` a `bool`?